### PR TITLE
code coverage with babel istanbul

### PR DESCRIPTION
**Pull request by trusktr**
Closes 

### Describe your changes:

replaces instanbul-intrumenter-loader with babel-plugin-istanbul. Vue `<script>`s are now covered, but not expressions in `<template>`

- [x] remove bad plugin/package and config
- [x] add new package and config
- [x] remove legacy-peer-deps option
- [ ] code coverage to be correct
- [ ] code coverage to work codecov.io
- [x] provide path towards coverage being usable in e2e tests
- [x] verify internal consumers still build and work fine

The previous instanbul-intrumenter-loader is outdated and abandoned with old Babel dependencies, so as our tooling progresses, it will be incompatible, and it crashes on any new syntax that all modern browsers already support without compiling or poyfilling those features.

This updates to use babel-plugin-istanbul which is maintained, so that it will stay up to date with Babel and latest syntaxes.

This also moves the converage instrumentation webpack config out of karma config into a separate webpack.coverage.js file, which can enable our other test environments besides Karma to also use it (not part of this PR).

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [ ] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [ ] Command line build passes?
* [ ] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

To test: run `npm run build:coverage` and verify that the output in `dist` has coverage instrumentation (you'll see weird statements injected all over the JS code, but if you see only your plain JS code and nothing strange, then it didn't work).

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@trusktr do you mind rebasing update-eslint on master and pullings those changes back into this PR?

**Review by unlikelyzero — DISMISSED** [review]

**Comment by trusktr** [comment]
hey @unlikelyzero , I see you approved. Are you okay with it not covering Vue code? Also I need to rebase this once both eslint PRs are merged. Then we can see if it'll work.

**Comment by unlikelyzero** [comment]
Yea that's fine for now

**Comment by akhenry** [comment]
Still needed in light of using code cov?

**Comment by shefalijoshi** [comment]
@trusktr Please resolve conflicts. Also, @unlikelyzero , @nikhilmandlik , @trusktr Do we still need this? Discuss and close as needed.

File: package.json [hunk 1]
```diff
@@ -9,6 +9,8 @@
     "@playwright/test": "1.19.2",
     "allure-playwright": "2.0.0-beta.15",
     "babel-eslint": "10.1.0",
+    "babel-loader": "^8.2.3",
```
Comment (unlikelyzero) [inline package.json:12]: let's pin these

File: karma.conf.js [hunk 2]
```diff
@@ -33,18 +33,18 @@ if (coverageEnabled) {
 module.exports = (config) => {
     const webpackConfig = require('./webpack.dev.js');
     delete webpackConfig.output;
-    if (coverageEnabled) {
```
Comment (unlikelyzero) [inline karma.conf.js:36]: if i'm reading this right, we need to remove all references to karma instrumentation, including any npm scripts

File: package.json (hunk 1 above)
Comment (unlikelyzero) [reply package.json:12]: I think we can also get rid of cross-env, istanbul-instrumenter-loader, karma-coverage-istanbul-reporter, and more

File: karma.conf.js (hunk 2 above)
Comment (trusktr) [reply karma.conf.js:36]: still left the karma istanbul reporter, as I assume it reads the instanbul output from babel-plugin-istanbul. Basically we only swapped istanbul-instrumenter-loader for babel-plugin-istanbul, but the rest stays the same as far as Karma, I believe.

File: package.json (hunk 1 above)
Comment (trusktr) [reply package.json]: pinned dependencies

I think karma-coverage-istanbul-reporter is still needed for karma to read and report the coverage

Comment (trusktr) [reply package.json]: We can't remove cross-env because it is still needed for `NODE_OPTIONS`

**Comment by trusktr** [comment]
I was unable to get instrumentation into the JS expressions inside `<template>`s of Vue components. I asked a couple question for ideas here:

- https://github.com/istanbuljs/babel-plugin-istanbul/issues/273
- https://github.com/JuniorTour/vue-template-babel-compiler/issues/38

File: babel.coverage.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const babelConfig = {
+    plugins: [['babel-plugin-istanbul', {
+        extension: ['.js', '.vue']
+        // include: ['./**/*.js', './**/*.vue*'],
+        // excludeNodeModules: false
```
Comment (unlikelyzero) [inline babel.coverage.js:5]: shouldn't we exclude node modules?

File: webpack.coverage.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const config = require('./webpack.dev');
```
Comment (unlikelyzero) [inline webpack.coverage.js:1]: Is there a reason we can't put this back into the `webpack.dev` config? I'm worried about managing another config

File: babel.coverage.js (hunk 3 above)
Comment (trusktr) [reply babel.coverage.js:5]: it defaults to `true` if not specirfied

File: webpack.coverage.js (hunk 4 above)
Comment (trusktr) [reply webpack.coverage.js:1]: I think it's a good idea not to make the dev build last longer when we don't need coverage during development (but I haven't measured the difference yet). We could also stick this inside karma conf just like the old config has istanbul-instrumentation-loader, but I think this is cleaner.

**Comment by jvigliotta** [comment]
@trusktr @unlikelyzero can you guys get together on this one?

**Review by unlikelyzero — APPROVED** [review]
We did a virtual walkthrough of the changes and potential impacts downstream.

**Pull request merged**